lib/pliny/librato/metrics/backend.rb

```diff
@@ -54,21 +57,13 @@ def start_timer
           end
         end
 
-        def start_counter
-          @counter = Thread.new do
-            loop do
-              msg = metrics_queue.pop
-              msg == POISON_PILL ? break : enqueue_librato(msg)
-            end
-          end
-        end
-
-        def enqueue_librato(msg)
-          sync { librato_queue.add(msg) }
-        end
-
         def flush_librato
-          sync { librato_queue.submit }
+          sync do
+            counter_cache.flush_to(librato_queue)
+            librato_queue.merge!(aggregator)
+            aggregator.clear
+          end
+          librato_queue.submit
```

Comment:
@cyx @joshwlewis The `Librato::Queue` is not thread safe and we have an interleaving now where this line can be called twice. When `#stop` is invoked and the timer loop calls `#flush_librato`.

Also as @mikehale pointed out above `Librato::Queue#merge!` will submit if the metric count threshold is hit.

I think all of these issues will need to be addressed as we're bound to run into them given enough time.